src/mainwindow/mainwindow.cpp

```diff
@@ -2960,9 +2960,17 @@ QString MainWindow::getStyleSheetSuffix() {
     return "fritzing";
 }
 
-void MainWindow::addToMyParts(ModelPart * modelPart)
+void MainWindow::addToMyParts(ModelPart * modelPart, QStringList & m_peAlienFiles)
 {
-    if (modelPart != NULL) m_binManager->addToMyParts(modelPart);
+    if (modelPart != NULL) {
+        foreach(QString pathToAddFromPe, m_peAlienFiles) {
+            // DebugDialog::debug(QString("addToMyParts adding  %1")
+            //.arg(pathToAddFromPe));
+            m_alienFiles << pathToAddFromPe;
+            m_alienPartsMsg = tr("Do you want to keep the imported parts?");
+        }
+    }
+	m_binManager->addToMyParts(modelPart);
 }
```

Comment:
The call to m_binManger now is outside of the null pointer check, was this intended?
This would be avoided by using const references (no null pointer check required, the method should not be called if it is expected to do nothing)

Reply:
@KjellMorgenstern  Sorry for the delay, I've been poking at this off and on. As I'm utterly new to c++ (I know c but not c++) that is taking some time. If I understand correctly, the m_peAlienFiles should change to just peAlienFiles (although I don't necessarily know why :-) ). The null pointer check is as intended (so that my code didn't do anything if there wasn't a modelPart present which I thought may occur under some case I don't understand), but from what you are saying should probably just be deleted if it won't get here without a modelPart. The other part I don't know about is if  the m_alienPartsMsg is correct. The data gets passed via the Qt IPC mechanism, but I'm not sure it is defined correctly (although it works and I don't see another way to get the definition from a single place in to both objects, there may be one). Once I figure out what needs to change the next challenge is going to be figuring out how to change the pull request (delete this one and submit a corrected one?) as I'm also new to (and usually baffled by) github.

Reply:
The pull request will pick up change automatically. Just create another commit and push it to your branch (3383_fix...)

It looks to me that the null pointer will only occur when out of memory, so it is not as relevant. In fact, it might sense to omit it and just crash with a proper error message, but that would be part of a refactoring